### PR TITLE
Update test_fused_layer_norm.py

### PR DESCRIPTION
**Pull request by Fuzzkatt**
Use parametrize and instantiate_device_type_tests to make tests cleaner.
Update torch.testing.assert_allclose to torch.testing.assert_close according to https://github.com/pytorch/pytorch/issues/61844.

### CODE REVIEW

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py [hunk 1]
```diff
@@ -8,291 +8,282 @@
 from apex.normalization import MixedFusedLayerNorm
 from apex.normalization import MixedFusedRMSNorm
 
+from torch.testing._internal import common_utils
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+from torch.testing._internal.common_device_type import onlyCUDA
```
Comment (crcrpar) [inline tests/L0/run_fused_layer_norm/test_fused_layer_norm.py:13]: Is this used somewhere? `instantiate_device_type_tests` would be enough
```suggestion

```

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py [hunk 2]
```diff
@@ -8,291 +8,282 @@
...
-                normalized_shape=self.normalized_shape, elementwise_affine=self.elementwise_affine).cpu()
-            self.module_cuda_ = FusedLayerNorm(
-                normalized_shape=self.normalized_shape, elementwise_affine=self.elementwise_affine).to(device="cuda", dtype=self.dtype)
-        else:
-            assert self.elementwise_affine
-            self.module_cpu_ = MixedFusedLayerNorm(
-                normalized_shape=self.normalized_shape).cpu()
-            self.module_cuda_ = MixedFusedLayerNorm(
-                normalized_shape=self.normalized_shape).to(device="cuda", dtype=self.dtype)
+from itertools import product
```
Comment (crcrpar) [inline tests/L0/run_fused_layer_norm/test_fused_layer_norm.py:15]: `itertools` is imported above, so could you remove either of them?

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py [hunk 3]
```diff
@@ -8,291 +8,282 @@
...
+    native = torch.nn.LayerNorm(
+        normalized_shape=normalized_shape, elementwise_affine=elementwise_affine
+    ).to(device="cuda", dtype=dtype)
+    fused = FusedLayerNorm(
+        normalized_shape=normalized_shape, elementwise_affine=elementwise_affine
+    ).cuda()
+    return native, fused
+
+
+def _prep_rms_layers(normalized_shape, elementwise_affine, dtype):
```
Comment (crcrpar) [inline tests/L0/run_fused_layer_norm/test_fused_layer_norm.py:27]: `dtype` doesn't seem to be used. comparing with `_prep_layers`, did you consider of casting `native` to `dtype`?

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py [hunk 4]
```diff
@@ -8,291 +8,282 @@
...
 
-    def test_autocast(self):
-        for (dtype, elementwise_affine) in itertools.product(autocast_dtypes, (True, False)):
-            self._run_test(dtype, elementwise_affine)
 
+instantiate_device_type_tests(TestFusedLayerNorm, globals(), only_for=("cuda",))
 
 if __name__ == "__main__":
-    unittest.main()
+    common_utils.run_tests()
```
Comment (crcrpar) [inline tests/L0/run_fused_layer_norm/test_fused_layer_norm.py:289]: could you add a new line at the end of file?
if you're using vscode, `"files.insertFinalNewline": true` to the setting file would do

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py (hunk 3 above)
Comment (crcrpar) [inline tests/L0/run_fused_layer_norm/test_fused_layer_norm.py:27]: nit-picking. it looks like these two are used once, it'd be fine to merge into the callsite or utlize these two in `_test_fused_layer_norm` and `_test_fused_rms_norm`.

**Review by crcrpar — COMMENTED** [review]
Could you check if there're unused imports using some tool e.g. flake8?

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py (hunk 3 above)
Comment (Fuzzkatt) [reply tests/L0/run_fused_layer_norm/test_fused_layer_norm.py]: Fixed

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py (hunk 1 above)
Comment (Fuzzkatt) [reply tests/L0/run_fused_layer_norm/test_fused_layer_norm.py]: Fixed

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py (hunk 2 above)
Comment (Fuzzkatt) [reply tests/L0/run_fused_layer_norm/test_fused_layer_norm.py:11]: Fixed

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py (hunk 4 above)
Comment (Fuzzkatt) [reply tests/L0/run_fused_layer_norm/test_fused_layer_norm.py]: Fixed

File: tests/L0/run_fused_layer_norm/test_fused_layer_norm.py (hunk 3 above)
Comment (Fuzzkatt) [reply tests/L0/run_fused_layer_norm/test_fused_layer_norm.py]: Added, also made corresponding fix to manual_rms_norm()

**Review by crcrpar — APPROVED** [review]
thanks, looks nice to me

**Pull request merged**